pkg/lib/index/indexer.go

```diff
@@ -0,0 +1,129 @@
+package index
+
+import (
+	"os"
+
+	"github.com/operator-framework/operator-registry/pkg/containertools"
+
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	defaultDockerfileName = "index.Dockerfile"
+	defaultImageTag = "operator-registry-index:latest"
+)
+
+// Indexer allows the creation of index container images from scratch or
+// based on previous index images
+type Indexer interface {
+	AddToIndex(AddToIndexRequest) error
+	DeleteFromIndex(DeleteFromIndexRequest) error
+}
+
+// IndexerImpl is a struct implementation of the Indexer interface
+type IndexerImpl struct {
```

Comment:
This is mostly a style thing, but I don't love `<interface> interface` and `<interface>Impl struct`, if you can think if something more descriptive.

For this one, I could imagine a a non-image-based implementation of this that operates on database files only. Then I would have `IndexerImpl` and `DatabaseFileIndexerImpl`, at which point why not have `ImageIndexer` and `DatabaseFileIndexer` so I can tell their intent apart?